cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowStatsSLAHttpHandlerTest.java

```diff
@@ -0,0 +1,117 @@
+
+    for (int i = 0; i < 10; i++) {
+      RunId workflowRunId = RunIds.generate();
+      store.setStart(workflowProgram, workflowRunId.getId(), RunIds.getTime(workflowRunId, TimeUnit.SECONDS));
+
+      RunId mapreduceRunid = RunIds.generate();
+      store.setWorkflowProgramStart(mapreduceProgram, mapreduceRunid.getId(), workflowProgram.getId(),
+                                    workflowRunId.getId(), mapreduceProgram.getId(),
+                                    RunIds.getTime(mapreduceRunid, TimeUnit.SECONDS), null, null);
+      store.setStop(mapreduceProgram, mapreduceRunid.getId(),
+                    TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()), ProgramRunStatus.COMPLETED);
+
+      // This makes sure that not all runs have Spark programs in them
+      if (i < 5) {
+        RunId sparkRunid = RunIds.generate();
+        store.setWorkflowProgramStart(sparkProgram, sparkRunid.getId(), workflowProgram.getId(),
+                                      workflowRunId.getId(), sparkProgram.getId(),
+                                      RunIds.getTime(sparkRunid, TimeUnit.SECONDS), null, null);
+        store.setStop(sparkProgram, sparkRunid.getId(),
+                      TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()), ProgramRunStatus.COMPLETED);
+      }
+      store.setStop(workflowProgram, workflowRunId.getId(),
+                    TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()), ProgramRunStatus.COMPLETED);
+
+      // This sleep is required so that the store is not overridden as the workflows can then
+      // potentially start at the same time causing conflicts
+      TimeUnit.SECONDS.sleep(1);
+    }
+
+    String request = String.format("%s/namespaces/%s/apps/%s/workflows/%s/statistics?start=%s&end=%s" +
+                                     "&percentile=%s&percentile=%s",
+                                   Constants.Gateway.API_VERSION_3, Id.Namespace.DEFAULT,
+                                   WorkflowApp.class.getSimpleName(), workflowProgram.getId(), "0", "now", "90", "95");
+
+    HttpResponse response = doGet(request);
+    WorkflowDataset.BasicStatistics basicStatistics =
+      readResponse(response, new TypeToken<WorkflowDataset.BasicStatistics>() { }.getType());
+    Assert.assertEquals(1, basicStatistics.getPercentileToRunids().get("90.0").size());
+    Assert.assertEquals(5,  Math.round(basicStatistics.getActionToStatistic().get(sparkName).get("count")));
+
+    request = String.format("%s/namespaces/%s/apps/%s/workflows/%s/statistics?start=%s&end=%s" +
+                              "&percentile=%s&percentile=%s",
+                            Constants.Gateway.API_VERSION_3, Id.Namespace.DEFAULT,
+                            WorkflowApp.class.getSimpleName(), workflowProgram.getId(), "now", "0", "90", "95");
+
+    response = doGet(request);
+    Assert.assertEquals(org.jboss.netty.handler.codec.http.HttpResponseStatus.BAD_REQUEST.getCode(),
```

Comment:
Shouldn't need to be fully qualified. Can just import HttpResponseStatus, right?